express-notes-api/index.js

```diff
@@ -0,0 +1,130 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const jsonNotes = require('./data.json');
+
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (const eachNote in jsonNotes) {
+    const outer = jsonNotes[eachNote];
+    for (const key in outer) {
+      notesArray.push(outer[key]);
+    }
+  }
+  res.json(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const idNum = parseInt(req.params.id);
+  var flag;
+  if (idNum < 0) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else if (idNum > 0) {
+    for (const eachNote in jsonNotes) {
+      const notesNum = jsonNotes[eachNote];
+      for (const x in notesNum) {
+        var noteId = notesNum[x];
+        if (idNum === noteId.id) {
+          res.status(200).send(noteId);
+          flag = true;
+        }
+      }
+    }
+
+  } if (!flag) {
+    res.status(404).send({ error: `cannot find note with id ${idNum}` });
+  }
+
+});
+
+app.use(express.json());
+
+app.post('/api/notes', (req, res) => {
+  if (req.body.constructor === Object && Object.keys(req.body).length === 0) {
+    res.status(400).send({ error: 'content is a required field' });
+  } else if (req.body.constructor === Object && Object.keys(req.body).length !== 0) {
```

Comment:
Since you are primarily concerned with the existence of a `content` property, you can check for its existence. It's possible that the object has a different property other than `content`, which will fool your code.

```js
if (!req.body.content) {
```